lib/bolt/inventory/inventory.rb

```diff
@@ -209,16 +210,43 @@ def create_target_from_hash(data)
         end
 
         if (aliases = new_target.target_alias)
+          @groups.insert_alia(new_target.name, aliases)
+        end
+
+        new_target
+      end
+
+      def validate_target_from_hash(target)
+        groups = Set.new(group_names)
+        targets = target_names
+
+        # Make sure there are no group name conflicts
+        if groups.include?(target.name)
+          raise ValidationError.new("Target name #{target.name} conflicts with group of the same name", nil)
+        end
+
+        # Validate any aliases
+        if (aliases = target.target_alias)
           aliases = [aliases] if aliases.is_a?(String)
```

Comment:
This was previously a helpful step because we reused this variable to pass to `insert_alia`. That's not the case anymore, so this conversion isn't helpful. I think we can remove this line and just check `unless aliases.is_a?(Array) || aliases.is_a?(String)` on the line below.